### ISSUE TITLE
EU servers are full of russian players.

### ISSUE THREAD

**Issue by joky21** [ValveSoftware/Dota2-Gameplay#3932]
What I do: search game on EU WEST and EU EAST servers. English language is selected as a language of in-game communication. Steam, Dota UI language is English.
What I get: 11 games in a row with 6-7 russians.

All matches are lost because of miscommunication. I dont understand them and they don't want to speak English. When I ask why do you play here? They say that I should shut up and die. 

It isn't a question of language. I cant just click a mute button. BUT, they are aggressive. When I do something they don't like, they start to take a bad tone. Insult on national grounds and so on.

Question: Is it possible to avoid them at all? If not, I'll just delete this game.

**Comment by davefojtik** [comment]
You can't. The language and region controls in EU are for decoration purpose only. Same as this repo. If you don't have 4 dota friends save yourself the trouble and do something healthier :)

**Comment by XVR-dev** [comment]
My friends and I abandoned Dota for this reason several month ago, and planning on leaving Steam entirely and moving to Epic Store. Valve still supports russians despite everything they do, and completely ignoring their devastating influence on game and it's community, and not even mentioning the fact that russia is a terrorist state who actually openly performs genocide right now. But it seems those bloody dirty rubles doesn't smell for Valve. Really disappointed with Valves position on russia and russians.

**Comment by davefojtik** [comment]
[@XVR-dev](https://github.com/XVR-dev) What do you mean by `Valve still supports Russians`?
The problem discussed here is about how the game system reacts to the fact that people are behaving wrong and are not speaking the language they've chosen. And it does not matter if the player that does it comes from Russia, Ukraine (_yes, it happened to me quite a lot_), the USA or Santa's north pole base. It can't even matter because geolocation is irrelevant today with VPNs.

I understand you have difficulties now, and you possibly didn't even want to be a part of this already rather embarrassing ongoing cold war. But convincing all the companies and people in the rest of the world everywhere you can that they should start hating one nation isn't going to help you solve it.

**Comment by XVR-dev** [comment]
@davefojtik 

I agree that what I wrote is not the direct answer to the question, but what I've mentioned is there is a bigger problem that comes not from some bug in Dota code, it comes from a bug in Valve's policy itself. As @joky21 mentioned in OP, the problem is that russians basically occupied other dota regions and they are aggressive and don't want to at very least comply with the community guidelines of common sense of speaking the language you've selected. It cannot be fixed in code, it can be only fixed by changing Valve's policy towards people who are not only actively disrupting and corrupting the community, but who, by common sense should not be allowed on the platform in general for being a f-ing actual terrorist. You say it doesn't matter where the person is from, and you are actually right, my whole point is that aggression should not be tolerated not in real life nor in online communities, no matter what is the source of this aggression. But, the people who are world's most open aggressors and terrorists, who actively trying to occupy other countries by killing and torturing innocent people and the people who are most known for their aggression and disrespect and disobedience of rules in online communities, I would guess, **_obviously, by pure coinsidence_**  happens to be the same people, and those happens to be russians nowdays. 

> But convincing all the companies and people in the rest of the world everywhere you can that they should start hating one nation isn't going to help you solve it.

Really hurts to hear that position. It just shows how a lot of western people still wildly out of touch what's happening in Ukraine and the basic history of eastern Europe in general, and who are russians by their nature, and the sole fact that it is still needs to be proven and some companies still needs to be convinced that it's not ok to work with literal terrorist state and furthermore actively whitewashing them, is truly horrifying. Tens of thousands of _civilians_ already died in Ukraine by russian hands and people are still violently murdered and tortured right now while I'm explaining it to you, and it's not the russian government who are killing and torturing those people it's the russian people themselves. When I say that the same guy who was yelling at you in voice chat, could, in a month, be torturing people in basements of occupied Ukrainian territories, believe me I'm literally not joking or overexaggerate this is what actually is happening here right now in 2022. 
The big problem is since you (the western people) don't understand russian language, you cannot believe us, those  who understand it, and you cannot believe that this whole country, like 90% of it's citizens **_actively_** supports russian aggression, and the rest are just passive imperialists who say to westerns that they're against the war and stuff, but what you don't hear is their answers to the right questions asked by the people who know what to ask, and just believe me at this point, there are no such thing as a "good" or "innocent" russian. And it's not because "I just hate them because I want to hate them", it's because they are actual terrorists and aggressors and the civilized world should unite against such aggression, and again , it could be any other country/nation (not really, but let's assume it really could) who would behave like this and openly be terrorist, and I would say the same stuff about that country, but, it happened to be, that modern terrorists are russians, and therefore they should not be tolerated and should be exiled from any civilized communities and discussions. And no sane company should do business with terrorists, and therefore not to deal with consequences of allowing such people in it's community. Cause it should be obvious that people who think it's a great idea to invade other country and perform thousands of war crimes, and killing civilians, are not going to obey the rules of civilized community.

**Comment by davefojtik** [comment]
@XVR-dev You're right. What you write here is not related to the question, nor anything you would discuss on this platform (apart from this failed repo). GitHub is mainly for software developers. Amazing things you use every day are being collaborated on here. The fact that based on your profile history you've made a GitHub account just to talk politics here does not make you behave any better than the trolls in dota. And does not kinda follow GitHub guidelines.

If you need someone to talk to about this search on different platforms. Try social media groups, discord servers, forums or sites focused on politics. I wish you good luck.

**Comment by XVR-dev** [comment]
It seams to me you've completely missed the context of what I've said. All of this was exactly the reasoning for the problem with dota which was described in original post, and why we have it in the first place, and why Valve will not fix it, so I consider it pretty relevant since as a result, players with 10+ years of dedication to dota completely leaving the community because of this issue. And that most of western players are totally ignorant/chill about it because of the lack of information. That's what the context is. 

And there's no need to arrogantly "explain to me what a github is" i'm pretty much well aware of what it is, after several years of software development.
It may come to you as a secret, but not all people use github as their main platform for dev work, there are some other platforms out there, and not all job-accounts are public, and people can have more that one account, and besides that, not all people like to use public repos for personal projects for that matter.

Anyways, this particular repo is not really for coders as you should be able to see, and not all issues here, like this one in particular, are about strictly code-related problems. I didn't come to some let's say "more technical" repo/topic and started talking politics out of the blue, I'm pretty much can read the title, to know where I am and what should I write.
I'm definitely not looking for a place to casually chat, I've had enough of this general topic during 9 years of ongoing war here.
And I'm definitely not looking for an advice on where, and on what topics should I communicate. 
And whilst I'm explaining the problem which the discussed **platform** obviously have according to the mentioned topic of the thread, you, for some reason, (which I really don't get), instead of adding something of use, sharing your thoughts on the topic of this discussion, just trying to undermine **me and my standpoint personally** by not arguing my actual point but by simply telling me "you cannot speak here, go somewhere else". Not very professional of you should I say... And not really compliant with github guidelines if you like to mention those, but who am I to to tell you this, right.

P.S.
Coming to some random thread about russians in dota and whilst completely ignoring the context of the thread telling people something like:
"GitHub is for devs and I'm a dev, and you are not, so you cannot speak on this platform for smart people like me, who create amazing products for you, peasants" - not really giving you any credibility and furthermore presenting you as not a particularly good dev/person in general, since being a good dev really implies having some level of soft-skills which would never lead someone to say something as arrogant as this, especially in a setting where you don't really know who are you talking to. smh.

**Comment by reutsharabani** [comment]
Is this ever going to get fixed?

I queue eu west and set language to English and nowdays I almost always have a strictly-russian speaking team.

I am pretty sure most people rather have longer queue time and be able to communicate with their team.

At least we should have it as an option.

**Comment by davefojtik** [comment]
@reutsharabani Do you have '`Strict Solo Role Queue Matchmaking`' option checked in the settings if you play solo? I guess you have bigger chance of everyone speaking english if they are playing solo than premade groups. 

But still the best thing you can do when the mm system sucks is to assemble your team before the match. From my experience you will get much better and balanced experience in premade even if ranks are quite far from each other. 

We could perhaps make some steam/discord groups or even opensource app for "3rd party" matchmaking like Faceit has but instead of playing on private servers you would play together on official ones 💡
